packages/theme-patternfly-org/helpers/asyncComponentFactory.js

```diff
@@ -0,0 +1,38 @@
+const React = require('react');
+
+const cache = {};
+
+function asyncComponentFactory(route, pageData) {
+  const { Component } = pageData;
+
+  class AsyncComponent extends React.Component {
+    state = {
+      isLoaded: false
+    };
+    static preload() {
+      return Component().then(res => {
+        cache[route] = res.default;
+        return res.default;
+      });
+    }
+    render() {
+      if (cache[route]) {
+        return React.createElement(cache[route]);
+      }
+      AsyncComponent.preload().then(() => this.setState({ isLoaded: true }));
+
+      // Simple loading state
+      return React.createElement('div',
+        { style: { height: '100vh' } },
```

Comment:
We have a Spinner component as well

Reply:
I can't import from patternfly in this file because it is `require()`d from a NodeJS context.

It'd take a little bit of refactoring but could be possible in the future. As it is you'll rarely see this thanks to preloading!